Src/FluentAssertions/Formatting/Line.cs

```diff
@@ -0,0 +1,146 @@
+using System;
+using System.Text;
+
+namespace FluentAssertions.Formatting;
+
+/// <summary>
+/// Represents a single line of output rendered through the <see cref="FormattedObjectGraph"/>.
+/// </summary>
+internal class Line
+{
+    /// <summary>
+    ///  The level of indentation at the time this line was created.
+    /// </summary>
+    private int indentation;
+
+    private string content;
+    private StringBuilder builder;
+
+    /// <summary>
+    /// If any whitespace was inserted at the beginning of the line without an <see cref="Anchor"/> knowing about it, this
+    /// will be the length of that whitespace so that any calls to <see cref="Insert"/> will be offset by this amount.
+    /// </summary>
+    private int whitespaceOffset;
+
+    /// <summary>
+    /// Creates an empty line with the specified indentation that will be applied when
+    /// actual fragments are added.
+    /// </summary>
+    public Line(int indentation)
+    {
+        this.indentation = indentation;
+        builder = new();
+    }
+
+    public Line(string content)
+    {
+        this.content = content.TrimEnd();
+    }
+
+    /// <summary>
+    /// Is used to close off the internal string builder.
+    /// </summary>
+    public void Flush()
+    {
+        if (builder is not null)
+        {
+            content = builder.ToString();
+            builder = null;
+        }
+    }
+
+    /// <summary>
+    /// Gets the length of the content, including any whitespace that was inserted at the beginning of the line.
+    /// </summary>
+    public int Length => builder?.Length ?? content.Length;
+
+    /// <summary>
+    /// Gets the length of the content without the offset of any whitespace that was inserted at the beginning of the line.
+    /// </summary>
+    public int LengthWithoutOffset => Length - whitespaceOffset;
+
+    public void Append(string fragment)
+    {
+        builder.Append(fragment);
+    }
+
+    public void InsertAtStart(string fragment)
+    {
+        if (builder is null)
+        {
+            content = content.Insert(0, fragment);
```

Comment:
🙃 like we do in `EnsureWhitespace()`

```suggestion
            content = fragment + content;
```